### ISSUE TITLE
[Media+Timeline] Expose media uploading progress through the timeline items

### ISSUE THREAD

**Issue by manuroe** [matrix-org/matrix-rust-sdk#1968]
Sister issue of #1733

Uploading media attachments should expose the loading progress through the related timeline item. That way we can present the user with that information, in the current context, within the timeline.

**Comment by jplatte** [comment]
We don't currently have a timeline item for media while the file is still being uploaded, so this is blocked on changing that (which is a while out). Still I'll assign myself since I've done most of the other timeline things as well as the general upload progress tracking.

**Comment by manuroe** [comment]
@jplatte you are right. We need local echo first. Those 2 features are in the same epic [File upload improvements](https://github.com/vector-im/internal-planning/issues/433) we will attack later.
